moVirt/src/main/java/org/ovirt/mobile/movirt/rest/OVirtClient.java

```diff
@@ -682,6 +670,65 @@ public void setupVersionHeader() {
         restClient.setHeader("Version", authenticator.getApiMajorVersion());
     }
 
+    private void setPersistentV3AuthHeaders() {
+        restClient.setHeader(SESSION_TTL, "120"); // 2h
+        restClient.setHeader(PREFER, "persistent-auth, csrf-protection");
+    }
+
+    private void resetClientSettings() {
+        restClient.setHeader(SESSION_TTL, "");
+        restClient.setHeader(PREFER, "");
+        restClient.setCookie(JSESSIONID, "");
+        restClient.setAuthentication(new HttpAuthentication() {
+            @Override
+            public String getHeaderValue() {
+                // empty authentication - e.g. not the basic one
+                return "";
+            }
+        });
+    }
+
+    private void updateClientBeforeCall() {
+        restClient.setHeader(FILTER, Boolean.toString(!authenticator.hasAdminPermissions()));
+        requestFactory.setCertificateHandlingMode(authenticator.getCertHandlingStrategy());
+        restClient.setRootUrl(authenticator.getApiUrl());
+    }
+
+    /**
+     * @param username username
+     * @param password password
+     * @return auth token depending on API version, if enforceBasicAuth is true then returns empty string
+     */
+    public String login(String username, String password) {
+        resetClientSettings();
+        updateClientBeforeCall();
+        restClient.setHttpBasicAuth(username, password);
+
+        Api api = restClient.loginV3();
```

Comment:
@jhernand thanx for the suggestion! Well, it would be convenient to be able to ask the API how to authenticate without this trial/error way, e.g. the /api would answer with the basic information about version without the need to login. On the other hand, not sure about security consequences...

But considering that even if something would be introduced to engine which would provide us with this information, moVirt still needs to support engines without this patch => we can not avoid the trial/error part.

@suomiy could you please change this part to work as Juan suggested?